### DISCUSSION TITLE
useless stack trace and error

### DISCUSSION THREAD

**Discussion by nicitaacom** [vercel/next.js#69603] [Help]
### Summary

![image](https://github.com/user-attachments/assets/4106557f-d9cf-41cf-ba59-62a42f334b76)

### Additional information

```js
How should I understand where error?
```

**Comment by Juneezee** [comment]
The error is this line in your `SecretKeyForm.tsx` file

```
Errors: Functions cannot be passed directly to Client Components unless you explicitly expose it by marking it with "use server"
```

Looks like you are using Server Action and trying to pass the Server Action function to your `SecretKeyForm` client component? In that case you need to mark your Server Action with `"use server"`.

See

1. https://react.dev/reference/rsc/server-actions#importing-server-actions-from-client-components
2. https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#client-components

**Reply by nicitaacom** [reply]
This discussion about useless errors like this one

Actually I'm not passing server action to client component as prop

This error was - because I passed function (or not function but something else) into server action (even with JSON.stringify - and then JSON.parse - it not work

By something else I mean
const supabaseServer = supabaseServerInstance()

I want jsut to see more usefull errors because this error was fixed in a way of commention all related code and then uncommenting it
